web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_msg_component.esm.js

```diff
@@ -0,0 +1,91 @@
+/** @odoo-module **/
+
+import {Component, onWillStart, useState} from "@odoo/owl";
+import {Dialog} from "@web/core/dialog/dialog";
+import {_t} from "@web/core/l10n/translation";
+
+export class ActWindowMessageDialog extends Component {
+    setup() {
+        this.state = useState({
+            buttons: [],
+        });
+        onWillStart(this.willStart);
+    }
+
+    async willStart() {
+        this.generateButtons();
+    }
+
+    _refreshWidget(env) {
+        const controller = env.services.action.currentController;
+        const state = env.services.router.current.hash;
+        const props = controller.props;
+        env.services.action.switchView(props.type, {resId: state.id});
+    }
+
+    generateButtons() {
+        var self = this;
+        const action = self.props.action;
+        const env = self.props.env;
+        if (action.close_button_title !== false) {
+            self.state.buttons.push({
+                name: action.close_button_title || _t("Close"),
+                click: () => {
+                    // Refresh the view before closing the dialog
+                    self._refreshWidget(env);
+                    self.props.close();
+                },
+                classes: "btn btn-default",
+            });
+        }
+        for (var i = 0; i < action.buttons.length; i++) {
```

Comment:
```suggestion
        for (var i = 0; action.buttons && i < action.buttons.length; i++) {
```

otherwise this fails if there are no buttons in the action dictionary